src/main.cpp

```diff
@@ -367,28 +381,28 @@ int main(int argc, char *argv[])
         handle_config_updated, NULL);
     core.init();
 
-    auto server_name = wl_display_add_socket_auto(core.display);
-    if (!server_name)
+    auto socket = choose_socket(core.display);
+    if (!socket)
     {
-        LOGE("failed to create wayland, socket, exiting");
+        LOGE("Failed to create wayland socket, exiting.");
 
         return -1;
     }
 
-    setenv("_WAYLAND_DISPLAY", server_name, 1);
-
-    core.wayland_display = server_name;
+    core.wayland_display = socket.value();
+    LOGI("Using socket name ", core.wayland_display);
     if (!wlr_backend_start(core.backend))
     {
-        LOGE("failed to initialize backend, exiting");
+        LOGE("Failed to initialize backend, exiting");
         wlr_backend_destroy(core.backend);
         wl_display_destroy(core.display);
 
         return -1;
     }
 
-    LOGI("running at server ", server_name);
-    setenv("WAYLAND_DISPLAY", server_name, 1);
+    setenv("_WAYLAND_DISPLAY", core.wayland_display.c_str(), 1);
```

Comment:
What's the point of this env var?

Reply:
I don't know lol, I copied that part from somewhere else and have not changed it since.

Reply:
Here is where this comes from: https://github.com/swaywm/rootston/blob/master/main.c#L46

However, sway doesn't set it.

Reply:
As emersion said on IRC, this is not really necessary, I removed it.